File: wasm/rushhour/src/widgets/mainWindow.cpp
#include "mainWindow.h"
#include "mainWidget.h"
#include "../levels.h"

MainWindow::MainWindow(QWidget *parent) :
    QMainWindow(parent)
{
    mainWidget = new MainWidget(this);
    mainWidget->setParent(this);

    QMenu *fileMenu = menuBar()->addMenu("File");
    level1Action = new QAction();
    level1Action->setText("Load Level 1");
    connect(level1Action, &QAction::triggered, this, &MainWindow::loadLevel1);
    level2Action = new QAction();
    level2Action->setText("Load Level 2");
    connect(level2Action, &QAction::triggered, this, &MainWindow::loadLevel2);
    level3Action = new QAction();
    level3Action->setText("Load Level 3");
    connect(level3Action, &QAction::triggered, this, &MainWindow::loadLevel3);
    fileMenu->addAction(level1Action);
    fileMenu->addAction(level2Action);
    fileMenu->addAction(level3Action);

    setCentralWidget(mainWidget);
    resize(700, 450);
    setWindowTitle("Rush Hour");
    setStatus("Loaded App");
    //setFixedSize(700, 450);
}

void MainWindow::loadLevel1()
{
    mainWidget->loadLevel(LEVEL_1_TEXT, "Level 1");
}

void MainWindow::loadLevel2()
{
    mainWidget->loadLevel(LEVEL_2_TEXT, "Level 2");
}

void MainWindow::loadLevel3()
{
    mainWidget->loadLevel(LEVEL_3_TEXT, "Level 3");
}

File: src/app/python/widgets/main_window.py
from PyQt5.QtWidgets import *
from PyQt5.QtGui import *
from PyQt5.QtCore import *
from .main_widget import MainWidget

class MainWindow(QMainWindow):
    def __init__(self, parent=None):
        super().__init__(parent)

        # widgets
        self.mainWidget = MainWidget(self)
        self.statusLabel = QLabel(self)

        self.setCentralWidget(self.mainWidget)
        self.resize(700, 450)
        self.setWindowTitle("Rush Hour")
        self.setWindowIcon(QIcon(":icon.png"))
        self.setStatus("Loaded App")
        self.setFixedSize(700, 450)
    
    def setStatus(self, status):
        self.statusBar().showMessage(str(status))
